hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -224,6 +224,24 @@ object LowerTableIR {
             Subst(newRow, BindingEnv(env, scan = Some(env)))
           })
 
+        case TableGroupWithinPartitions(child, groupedStructName, n) =>
+          val loweredChild = lower(child)
+          val keyFields = FastIndexedSeq(loweredChild.partitioner.kType.fieldNames: _*)
```

Comment:
I think the partitioner could have a longer key than the IR key. Use `child.typ.keyType`

Reply:
Agreed and addressed